plugin/src/main/java/com/denizenscript/denizen/objects/properties/item/ItemPotion.java

```diff
@@ -79,72 +44,33 @@ public static MapTag effectToMap(PotionEffect effect) {
     }
 
     public ListTag getMapTagData(boolean includeExtras) {
-        ListTag result = new ListTag();
-        MapTag base = new MapTag();
-        if (isPotion()) {
-            PotionMeta meta = getPotionMeta();
+        ListTag result = new ListTag(getCustomEffects(), ItemPotion::effectToMap);
+        if (getItemMeta() instanceof PotionMeta potionMeta) {
+            MapTag base = new MapTag();
             if (NMSHandler.getVersion().isAtLeast(NMSVersion.v1_20)) {
-                base.putObject("base_type", new ElementTag(Utilities.namespacedKeyToString(meta.getBasePotionType().getKey())));
-            }
-            if (includeExtras) { // TODO: Eventually remove these 3.
-                base.putObject("type", new ElementTag(meta.getBasePotionData().getType()));
-                base.putObject("upgraded", new ElementTag(meta.getBasePotionData().isUpgraded()));
-                base.putObject("extended", new ElementTag(meta.getBasePotionData().isExtended()));
-            }
-            if (meta.hasColor()) {
-                base.putObject("color", BukkitColorExtensions.fromColor(meta.getColor()));
+                base.putObject("base_type", new ElementTag(Utilities.namespacedKeyToString(potionMeta.getBasePotionType().getKey()), true));
+            }
+            if (includeExtras) { // TODO: Eventually remove these 4
+                base.putObject("type", new ElementTag(potionMeta.getBasePotionData().getType()));
+                base.putObject("upgraded", new ElementTag(potionMeta.getBasePotionData().isUpgraded()));
+                base.putObject("extended", new ElementTag(potionMeta.getBasePotionData().isExtended()));
+                if (potionMeta.hasColor()) {
+                    base.putObject("color", BukkitColorExtensions.fromColor(potionMeta.getColor()));
+                }
             }
-            result.addObject(base);
-        }
-        for (PotionEffect effect : getCustomEffects()) {
-            result.addObject(effectToMap(effect));
+            result.addObject(0, base);

```

Comment:
This is a really weird way of doing this - why are you injecting the first entry in at the end? It's much clearer code when list additions are done in the order they're meant to be (not to mention more performant - add-at-zero is an expensive op)

Reply:
Fair enough, I guess just initializing the list to the correct size and adding stuff as usual would be the better - should be done